### PR TITLE
herbstluftwm: fix unable to be started

### PR DESCRIPTION
**Pull request by B4rc1**
## Description of changes

After updating I could not start herbstluftwm since the X session would exit immediately. After a lot of trouble shooting I found that this commit caused the issue for me: https://github.com/NixOS/nixpkgs/commit/7cfeb768e5f2a160075d8736101444c10f1c5ea6
I am not sure why this change was introduced so I kept the changes and looked to how the implementation of other window manager modules, namely i3 handle this: https://github.com/NixOS/nixpkgs/blob/50aa30a13c4ab5e7ba282da460a3e3d44e9d0eb3/nixos/modules/services/x11/window-managers/i3.nix#L58-L65

So i did the same for herbstluftwm and now it works for me :)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ThibautMarty** [comment]
Thanks.
Please run the tests, though.

**Comment by pjones** [comment]
I can confirm that this change fixes the issue and allows the test to pass.

**Comment by fdietze** [comment]
I can confirm that this PR fixes both use-cases. Thank you!

```
desktopManager.xfce.enable = true;
windowManager.herbstluftwm.enable = true;
```

And then try to login with those sessions:

- none+herbstluftwm
- xfce+herbstluftwm

both work as expected.


It fixes the bug introduced in: https://github.com/NixOS/nixpkgs/pull/237364

**Comment by fdietze** [comment]
@SuperSandro2000 you merged my previous PR (https://github.com/NixOS/nixpkgs/pull/237364), which introduced a bug. Could you please also merge this PR which fixes the bug?

File: nixos/modules/services/x11/window-managers/herbstluftwm.nix [hunk 1]
```diff
@@ -33,7 +33,10 @@ in
             (cfg.configFile != null)
             ''-c "${cfg.configFile}"''
             ;
-        in "${cfg.package}/bin/herbstluftwm ${configFileClause} &";
+        in ''
+          ${cfg.package}/bin/herbstluftwm ${configFileClause} &
+          waitPID=$!
+          '';
```
Comment (SuperSandro2000) [inline nixos/modules/services/x11/window-managers/herbstluftwm.nix:39]:
```suggestion
        '';
```

**Comment by kragniz** [comment]
I've tested this change locally, it fixes herbstluftwm for me

**Pull request merged**